### PR TITLE
Move parameterized logging from `LoggingUtil` to `ILogger`

### PR DESCRIPTION
**Pull request by JackPGreen**
When I first started working with the codebase, I wondered if the `ILogger` supported parameterized logging (based on previous log4j experience) - but the interface offered nothing, so I didn't think it did. 

Later I found that this is possible, but with the `LoggingUtil` utility class.

Changes:
- Migrated `LoggingUtil` methods into `ILogger` so that they are more discoverable
- Delegated `LoggingUtil` functionality to directly call `ILogger` implementations
- Deprecated `LoggingUtil`
- Added Javadoc
- Remove duplication of methods defined in `ILogger` interface already declared in `TpcLogger` parent

This is a step towards addressing https://github.com/hazelcast/hazelcast/issues/25836

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/logging/ILogger.java [hunk 1]
```diff
@@ -205,17 +35,68 @@ public interface ILogger extends TpcLogger {
...
+        }
+    }
+
+    /**
+     * Logs to {@link #finest(String)} using a lazily evaluated {@code template} {@link String} with arguments, formatted using
+     * {@link String#format(String, Object...)}
+     *
+     * @since 5.4
+     */
+    default void logFinest(String template, Object arg1) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/logging/ILogger.java:76]: I 'd rather skip "log" in the method name to match existing methods' naming: `finest`, `fine` etc

Comment (JackPGreen) [reply hazelcast/src/main/java/com/hazelcast/logging/ILogger.java]: a547aba8cadcc6ed16fc3f41e1d1f2415cce159f

**Comment by JackPGreen** [comment]
run-lab-run

**Comment by pveentjer** [comment]
Note: the ILogger is part of the public API (unfortunately). So we need to be extremely careful to not break anything.

**Review by vbekiaris — APPROVED** [review]

**Review by JamesHazelcast — APPROVED** [review]
Thanks for the cleanup @JackPGreen 👍

**Pull request merged**